agents/vantagePro2_agent/vantage_pro2_agent.py

```diff
@@ -0,0 +1,166 @@
+if not on_rtd:
+    from ocs import ocs_agent, site_config
+    from ocs.ocs_twisted import TimeoutLock
+
+
+class VantagePro2Agent:
+
+    def __init__(self, agent, port="/dev/ttyUSB0", freq_in=2):
+        self.active = True
+        self.agent: ocs_agent.OCSAgent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+
+        self.port = port
+        self.module: Optional[VantagePro2] = None
+
+        self.freq = freq_in
+
+        self.initialized = False
+        self.take_data = False
+
+        # Registers Temperature and Voltage feeds
+        agg_params = {
+            'frame_length': 60,
+        }
+        self.agent.register_feed('weather_data',
+                                 record=True,
+                                 agg_params=agg_params)
+
+    # Task functions.
+    def init_VantagePro2_task(self, session, params=None):
+        """
+        Perform first time setup of the Weather Monitor Module.
+
+        Args:
+            params (dict): Parameters dictionary for passing parameters to
+                task.
+
+        """
+        if params is None:
+            params = {}
+
+        auto_acquire = params.get('auto_acquire', False)
+
+        if self.initialized:
+            return True, "Already Initialized Module"
+
+        with self.lock.acquire_timeout(0, job='init') as acquired:
+            if not acquired:
+                self.log.warn("Could not start init because "
+                              "{} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('starting')
+
+            self.module = VantagePro2(self.port)
+            print("Initialized Vantage Pro2 module: {!s}".format(
+                self.module))
+
+        self.initialized = True
+
+        # Start data acquisition if requested
+        if auto_acquire:
+            self.agent.start('acq')
+
+        return True, 'Vantage Pro2 module initialized.'
+
+    def start_acq(self, session, params=None):
+        """
+        Method to start data acquisition process.
+
+        Args:
+            sample_freq (int):
+                Frquency at which weather data is sampled.
+                Defaults to sample/2 seconds, the highest frequency possible.
+
+        """
+        time.sleep(2)
+        if params is None:
+            params = {}
+
+        sample_freq = params.get('freq')
+        # If loops is None, use value passed to Agent init
+        if sample_freq is None:
+            sample_freq = self.freq
+
+        with self.lock.acquire_timeout(0, job='acq') as acquired:
+            if not acquired:
+                self.log.warn("Could not start acq because {} is already running"
+                              .format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('running')
+
+            self.take_data = True
+
+            while self.take_data:
```

Comment:
I suggest using the [Pacemaker](https://ocs.readthedocs.io/en/latest/api.html?highlight=Pacemaker#ocs.ocs_twisted.Pacemaker) class to have even quantized sampling.